drracket/drracket/private/frame.rkt

```diff
@@ -872,10 +873,9 @@
     (define sorted-items (sort items cmp))
     
     (define (different-slots? item-key next-item-key)
-      (or (not (= (quotient item-key 100)
-                  (quotient next-item-key 100)))
-          (not (= (sgn item-key)
-                  (sgn next-item-key)))))
+      (not (and (= (quotient item-key 100)
+                   (quotient next-item-key 100)) (= (sgn item-key)
+                                                    (sgn next-item-key)))))
```

Comment:
I don't think I like this one. I can see why it was done and it does seem like a good idea in some cases, but I think that the slots are different in two situations; either the one thingy is different or the other thingy was different. Renaming the function to `same-slots?` and putting a `not` at the callsite works, however!

Reply:
The rules related to De Morgan's laws have felt a little confusing to me in some cases too. I think they're not worth keeping.

Reply:
Does it make sense to have some kind of "boolean normalization" rules that search for the smallest equivalent expression and then, if it is especially smaller than the current one, suggest it?

Reply:
That is, I have found that thinking at that level (boolean identities) has been helpful even if I don't know how it is going to turn out in the end; seeing how it turns out often helps me decide if I like it or not.

Reply:
I think I'd rather not even try to solve it with automation. There's bigger fish to fry, and making sure it never makes things worse seems very difficult. The ideal suggestions for Resyntax to make are those that most people consider clearly and significantly better.